src/audio/copier/copier.c

```diff
@@ -536,21 +463,59 @@ static int copier_copy_to_sinks(struct copier_data *cd, struct comp_dev *dev,
 	return ret;
 }
 
-static int copier_module_copy(struct copier_data *cd, struct comp_dev *dev)
+static int copier_module_copy(struct processing_module *mod,
+			      struct input_stream_buffer *input_buffers, int num_input_buffers,
+			      struct output_stream_buffer *output_buffers, int num_output_buffers)
 {
+	struct copier_data *cd = module_get_private_data(mod);
 	struct comp_buffer __sparse_cache *src_c;
 	struct comp_copy_limits processed_data;
-	struct comp_buffer *src;
-	int ret;
+	int i;
 
-	src = list_first_item(&dev->bsource_list, struct comp_buffer, sink_list);
-	src_c = buffer_acquire(src);
+	if (!num_input_buffers || !num_output_buffers)
```

Comment:
this is also CI or QB fix?
for module case:
dev->ipc_config.type == SOF_COMP_NONE

can we return at:

	if (num_output_buffers == 1) {
		module_single_sink_setup(dev, source_c, sinks_c);
		if (sinks_c[0]->sink->state != dev->state) {
			num_output_buffers = 0;
			buffer_release(sinks_c[0]);
                       goto out;
		}
	} else if (num_input_buffers == 1) {
		module_single_source_setup(dev, source_c, sinks_c);
		if (source_c[0]->source->state != dev->state) {
			num_input_buffers = 0;
			buffer_release(source_c[0]);
                        goto out;
		}
	} else {
		ret = -EINVAL;
		goto out;
	}

Reply:
Type will be NONE for every component.not just the module copier except for host or dai copier

Reply:
possible to remove line 475-476 with above code? I added "goto out" for each zero case in module_adapter_audio_stream_type_copy

Reply:
@btian1 no unfortunately not. It is up to the module to decide how it wants to handle this case. For example, the mixin will choose to skip processing altogether whereas the mixout will need to produce silence.

Reply:
then mixin and mixout should call: module_adapter_audio_stream_type_copy separately right? still different num_input_buffers and num_output_buffers? I just want make module process as simple as it can.